part_2/js/script.js

```diff
@@ -0,0 +1,149 @@
+        }else{
+            const container = document.getElementById(side);
+
+            let loadingTextContainer = document.createElement('p');
+            loadingTextContainer.innerHTML=loadingText;
+            container.appendChild(loadingTextContainer);
+
+            let loadingImageContainer = document.createElement('div');
+            loadingImageContainer.classList.add('loading');
+            loadingImageContainer.classList.add('centered');
+            container.appendChild(loadingImageContainer);
+        }
+    }
+
+    function onChange(evt){
+        console.log(evt.target.value);
+        //Begin step 2
+        clearContainer('right');
+        if(evt.target.value!=''){//not the first?
+            fetch(url)//get employee
+            .then(response => response.json())
+            .then((employees)=>{
+                displayEmployee(employees[evt.target.value-1]);
+            });
+        }
+    }
+
+    function displayEmployee(emp){//helper
+        const br=document.createElement('br');
+        console.log(emp);
+        let div = document.createElement('div');//make div
+        let h1 = document.createElement('h1');//make h1
+        h1.innerHTML= emp.first_name+' '+emp.last_name;
+        div.appendChild(h1);
+        div.appendChild(br);
+        let h2 = document.createElement('h2');//make h2
+        h2.innerHTML=emp.department.name;
+        div.appendChild(h2);
+        div.appendChild(br);
+        let pa=document.createElement('p');//make p
+        pa.appendChild(document.createTextNode(
+            `Salary: ${emp.annual_salary}
+\nHire date: ${emp.hire_date}`));//I have no idea why these new line things aren't working *or* why appending h tags just adds the text and nothing else
+        div.appendChild(pa);
+        //Begin step 4
+        let a=document.createElement('a');//make anchor
+        a.appendChild(document.createTextNode('View Department'));
+        a.href="#"//click?
+        a.id="department"
+        div.appendChild(a);
+        div.appendChild(br);
+    //End step 4
+
+        //Begin step 3
+        let img=document.createElement("img");//make image
+        img.src=emp.image_filename;
+        div.appendChild(img);
+        div.appendChild(br);
+        //End step 3
+
+        right.innerHTML=div.innerHTML;
+        //End step 2
+
+        //Begin step 4
+        document.getElementById('department').addEventListener('click',()=>{
+            clearContainer('right');
+            displayEmployee(emp);
+            let h4=document.createElement('h4').appendChild(document.createTextNode('Department: '+emp.department.name));
+            let h5=document.createElement('h5').appendChild(document.createTextNode('Employees'));
+
+            right.appendChild(h4);
+            right.appendChild(br);
+            right.appendChild(h5);
+            right.appendChild(br);
+            depatrmentEmployees(emp)
+        });
+        alert(right.innerHTML);
```

Comment:
I'd rather not see these as alerts - just display on the page.

Reply:
That one I missed during cleanup